src/mscorlib/src/System/Span.cs

```diff
@@ -614,16 +614,18 @@ internal static unsafe void CopyTo<T>(ref T destination, ref T source, int eleme
                 {
                     fixed (byte* pSource = &Unsafe.As<T, byte>(ref source))
                     {
-#if BIT64
-                        Buffer.Memmove(pDestination, pSource, (ulong)elementsCount * (ulong)Unsafe.SizeOf<T>());
-#else
-                        Buffer.Memmove(pDestination, pSource, (uint)elementsCount * (uint)Unsafe.SizeOf<T>());
-#endif
+                        Buffer.Memmove(pDestination, pSource, (nuint)elementsCount * (nuint)Unsafe.SizeOf<T>());
                     }
                 }
             }
             else
             {
+                if ((nuint)elementsCount * (nuint)Unsafe.SizeOf<T>() >= 128 &&
+                    RuntimeImports.RhCopyMemoryWithReferences(ref destination, ref source, elementsCount))
```

Comment:
I do not think we need special casing for small sizes here. The FCall should be fine for all case - if it is called directly from here; and avoids unnecessary layers internally.